common/src/java/com/zimbra/common/soap/W3cDomUtil.java

```diff
@@ -91,29 +81,58 @@ private W3cDomUtil() {}
             }
     };
 
+    public static DocumentBuilderFactory makeDocumentBuilderFactory() throws ParserConfigurationException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        dbf.setNamespaceAware(true);
+        dbf.setIgnoringComments(true);
+        // protect against recursive entity expansion DOS attack and perhaps other things
+        dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        try {
+            // XXE attack prevention
+            dbf.setFeature(Constants.DISALLOW_DOCTYPE_DECL, true);
+            dbf.setFeature(Constants.EXTERNAL_GENERAL_ENTITIES, false);
+            dbf.setFeature(Constants.EXTERNAL_PARAMETER_ENTITIES, false);
+            dbf.setFeature(Constants.LOAD_EXTERNAL_DTD, false);
+            dbf.setXIncludeAware(false);
```

Comment:
I've done testing by verifying features on factory object in unit tests. Can you guide me on how to test this?